crates/matrix-sdk/src/room/mod.rs

```diff
@@ -614,15 +616,22 @@ impl Room {
     }
 
     /// Gets additional context info about the client crypto.
-    pub async fn crypto_context_info(&self) -> CryptoContextInfo {
-        CryptoContextInfo {
-            device_creation_ts: match self.client.encryption().get_own_device().await {
-                Ok(Some(device)) => device.first_time_seen_ts(),
-                // Should not happen, there will always be an own device
-                _ => MilliSecondsSinceUnixEpoch::now(),
-            },
-            is_backup_configured: self.client.encryption().backups().state()
-                == BackupState::Enabled,
+    pub async fn crypto_context_info(&self) -> Option<CryptoContextInfo> {
+        #[cfg(feature = "e2e-encryption")]
```

Comment:
Given that we're only going to call this if the `e2e-encryption` feature is enabled (I think?) you could instead do:

```suggestion
        #[cfg(feature = "e2e-encryption")]
        pub async fn crypto_context_info(&self) -> CryptoContextInfo {
           ...
        }
```

(and likewise in the `RoomDataProvider` trait)

... which would mean that the `CryptoContextInfo` type could stay put in `matrix-sdk-crypto`.

Reply:
[revert CryptoContetInfo optional as the ui crate requires e2ee](https://github.com/matrix-org/matrix-rust-sdk/pull/4275/commits/85b907e6a415a5206c466dc012ed93dde72b2464) 
 then https://github.com/matrix-org/matrix-rust-sdk/pull/4275/commits/0f87ea9d3eb58836ccd080092947dd971d992190